recipe/meta.yaml

```diff
@@ -38,129 +42,142 @@ build:
+    - {{native}}gsl
+    - {{native}}libtiff >=4.0.3,<4.0.8   # [win]
+    - {{native}}libtiff                  # [not win]
+    - {{native}}libxml2                  # [win]
+    - {{native}}libxml2                  # [not win]
+    - {{native}}libssh2                  # [not win]
+    - {{native}}krb5                     # [not win]
+    - {{native}}openblas-devel >=0.2.20  # [linux or osx]
+    - {{native}}openblas                 # [win]
 
   run:
-    - readline                 # [not win]
-    - ncurses                  # [not win]
-    - jpeg                     # [not win]
-    - curl                     # [not win]
-    - xz                       # [not win]
-    - libpng                   # [not win]
-    - cairo                    # [not win]
-    - libssh2                  # [not win]
-    - openssl                  # [not win]
-    - krb5                     # [not win]
-    - {{native}}bzip2 1.0.*    # [not osx]
-    - {{native}}libjpeg-turbo  # [win]
-    - {{native}}libiconv       # [win]
-    - {{native}}gmp 6.1.*      # [win]
-    - {{native}}fftw           # [win]
-    - {{native}}xz             # [win]
-    - {{native}}mpfr 3.1.*     # [win]
-    - {{native}}libsndfile     # [win]
-    - {{native}}bwidget        # [win]
-    - {{native}}tktable        # [win]
-    - {{native}}libpng         # [win]
-    - pango                    # [not win]
-    - icu                      # [not win]
-    - pcre                     # [not win]
-    - {{native}}pcre           # [win]
-    - {{native}}tk             # [win]
-    - tk                       # [not win]
-    - zlib                     # [not win]
-    - {{native}}zlib           # [win]
-    - {{native}}gsl            # [win]
-    - gsl                      # [not win]
-    - libtiff                  # [not win]
-    - {{native}}libtiff        # [win]
-    - {{native}}libxml2 2.9.*  # [win]
-    - libxml2                  # [not win]
-    - xorg-libx11              # [linux]
-    - xorg-libxt               # [linux]
-    - xorg-libxrender          # [linux]
-    - xorg-libxext             # [linux]
-    - xorg-libxdmcp            # [linux]
+    - _r-mutex 1.* anacondar_1
+    # Needed by R CMD javareconf
+    - make                               # [not win]
+    - {{ compiler('fortran') }}          # [not win]
+    - {{ compiler('cxx') }}              # [not win]
+    - {{ compiler('c') }}                # [not win]
+    - llvm-openmp >=4.0.1                # [osx]
+    - {{native}}gcc-libs                 # [win]
+    - curl                               # [not win]
+    - {{native}}bzip2 1.0.*              # [win]
+    - {{native}}libjpeg-turbo            # [win]
+    - {{native}}libiconv                 # [win]
+    - {{native}}libuuid                  # [linux]
+    - {{native}}gmp                      # [win]
+    - {{native}}fftw                     # [win]
+    - {{native}}xz                       # [win]
+    - {{native}}mpfr                     # [win]
```

Comment:
This should be pinned.

Reply:
No they should not, we use run_exports.

Reply:
I am not going to infect my recipes with pinning.

Reply:
You are wrong here anyway, msys2 packages *never* need pinning, they use an epoch system instead.

Reply:
> You are wrong here anyway, msys2 packages never need pinning, they use an epoch system instead.

Can you explain this please?

Reply:
> Every msys2 package depends on an epoch package and that is versioned at the date when the upstream rolling distribution packages were converted to conda packages. That is the only version constraint here because that's how rolling distributions work.

Sure. How does the r-base package pin to that epoch package? Is it done by `run_exports`?

Reply:
Transitively each msys2 dependency depends on the same epoch version.

Reply:
Future-wise, R upstream are now based on MSYS2 (and are thus more up to date than us) so we may want to consider just bundling all their static libs into a single package and using that in the future. We'd also need to update our mingw-w64 toolchain (or provide the 'upstream' one, just for R).